pkg/view/component/signpost.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) 2021 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package component
+
+import (
+	"github.com/vmware-tanzu/octant/internal/util/json"
+)
+
+// Signpost is a component for a signpost
+// +octant:component
+type Signpost struct {
+	Base
+	Config SignpostConfig `json:"config"`
+}
+
+// SignpostConfig is the contents of a signpost
+type SignpostConfig struct {
+	// Component is the component that will be displayed.
+	Component Component `json:"component,omitempty"`
+	// Trigger is the component that will trigger the signpost.
+	Trigger Component `json:"trigger"`
+	// Message is the text that will be displayed.
+	Message string `json:"message"`
+	// Reverse is the order of the components.
+	Reverse bool `json:"reverse"`
+	// Column is the layout.
+	Column bool `json:"column"`
+	// Position of Signpost
+	Position Position `json:"position"`
+}
+
+type Position string
+
+const (
+	PositionTopLeft      Position = "top-left"
+	PositionTopMiddle    Position = "top-middle"
+	PositionTopRight     Position = "top-right"
+	PositionRightTop     Position = "right-top"
+	PositionRightMiddle  Position = "right-middle"
+	PositionRightBottom  Position = "right-bottom"
+	PositionBottomRight  Position = "bottom-right"
+	PositionBottomMiddle Position = "bottom-middle"
+	PositionBottomLeft   Position = "bottom-left"
+	PositionLeftBottom   Position = "left-bottom"
+	PositionLeftMiddle   Position = "left-middle"
+	PositionLeftTop      Position = "left-top"
+)
+
+func NewSignpost(c Component, t Component, m string) *Signpost {
+	so := &Signpost{
+		Base: newBase(TypeSignpost, nil),
+		Config: SignpostConfig{
+			Component: c,
+			Trigger:   t,
+			Message:   m,
+			Reverse:   false,
```

Comment:
Well, let say that you have a text and an icon, if you want the icon to be the trigger to show the signPost swapping will be a problem

Reply:
The only case I can think of where were need both a component and trigger is the text and icon combination. If this is true, maybe it is easier to make the message part of the icon component. Then this signpost component won't need a component and trigger with the reverse configuration going away.

Reply:
I was thinking in the decorator pattern here (build large component base on small ones), I don't really know if is necessary but I have seen it on forms (first image), again I don't know if the icon should be with the label and we should use the column layout (second image), but it will be awesome if we can have this discussions during planning

![image](https://user-images.githubusercontent.com/6991432/113296820-8ef8a200-92bf-11eb-86f7-019ac1413389.png)
![image](https://user-images.githubusercontent.com/6991432/113297233-fca4ce00-92bf-11eb-870c-897586b02359.png)

Reply:
done!